metrics-jersey2/src/main/java/com/codahale/metrics/jersey2/InstrumentedResourceMethodApplicationListener.java

```diff
@@ -126,19 +133,48 @@ public ExceptionMeterMetric(final MetricRegistry registry,
      * different response codes
      */
     private static class ResponseMeterMetric {
+        private static final Set<ResponseMeteredLevel> COARSE_METER_LEVELS = EnumSet.of(COARSE, ALL);
+        private static final Set<ResponseMeteredLevel> DETAILED_METER_LEVELS = EnumSet.of(DETAILED, ALL);
         public final List<Meter> meters;
+        private final Map<Integer, Meter> responseCodeMeters;
+        private final MetricRegistry metricRegistry;
+        private final String metricName;
+        private final ResponseMeteredLevel level;
 
         public ResponseMeterMetric(final MetricRegistry registry,
                                    final ResourceMethod method,
                                    final ResponseMetered responseMetered) {
-            final String metricName = chooseName(responseMetered.name(), responseMetered.absolute(), method);
-            this.meters = Collections.unmodifiableList(Arrays.asList(
+            this.metricName = chooseName(responseMetered.name(), responseMetered.absolute(), method);
+            this.level = responseMetered.level();
+            this.meters = COARSE_METER_LEVELS.contains(level) ?
+                    Collections.unmodifiableList(Arrays.asList(
                     registry.meter(name(metricName, "1xx-responses")), // 1xx
                     registry.meter(name(metricName, "2xx-responses")), // 2xx
                     registry.meter(name(metricName, "3xx-responses")), // 3xx
                     registry.meter(name(metricName, "4xx-responses")), // 4xx
                     registry.meter(name(metricName, "5xx-responses"))  // 5xx
-            ));
+            )) : Collections.emptyList();
+            this.responseCodeMeters = DETAILED_METER_LEVELS.contains(level) ? new ConcurrentHashMap<>() : Collections.emptyMap();
+            this.metricRegistry = registry;
+        }
+
+        public void mark(int statusCode) {
+            if (DETAILED_METER_LEVELS.contains(level)) {
+                getResponseCodeMeter(statusCode).mark();
+            }
+
+            if (COARSE_METER_LEVELS.contains(level)) {
+                final int responseStatus = statusCode / 100;
+                if (responseStatus >= 1 && responseStatus <= 5) {
+                    meters.get(responseStatus - 1).mark();
+                }
+            }
+        }
+
+        public Meter getResponseCodeMeter(int statusCode) {
```

Comment:
question: Does this method need to be `public` or would a more restrictive visibility scope also work?

Reply:
Changed to private